tests/unittest/_torch/test_disaggregated_single_gpu.py

```diff
@@ -0,0 +1,228 @@
+
+    # Context worker
+    worker_pytorch_configs.append(
+        PyTorchConfig(enable_overlap_scheduler=False,
+                      kv_cache_dtype="auto",
+                      use_cuda_graph=enable_cuda_graph))
+
+    # Generation worker
+    worker_pytorch_configs.append(
+        PyTorchConfig(enable_overlap_scheduler=generation_overlap,
+                      kv_cache_dtype="auto",
+                      use_cuda_graph=enable_cuda_graph))
+
+    kv_cache_configs = [KvCacheConfig(max_tokens=2048 * 8) for _ in range(2)]
+    model_names = [model_path(model) for _ in range(2)]
+    ranks = [0, 1]
+    worker_args = list(
+        zip(kv_cache_configs, worker_pytorch_configs, model_names, ranks,
+            [attention_dp_context, attention_dp_generation]))
+
+    port_name = MPI.Open_port()
+    MPI.Publish_name('my_port', port_name)
+
+    with MPIPoolExecutor(max_workers=2, env={"TRTLLM_USE_MPI_KVCACHE":
+                                             "1"}) as executor:
+        futures = []
+        try:
+            for worker_arg in worker_args:
+                future = executor.submit(worker_entry_point, *worker_arg)
+                futures.append(future)
+        except Exception as e:
+            print(f"Error in worker {worker_arg}: {e}")
+            raise e
+
+        try:
+            print("Launched all the workers.")
+            intercomm = MPI.COMM_SELF.Accept(port_name)
+
+            for _ in range(2):
+                intercomm.recv(tag=MPI_READY)
+                print("Received ready signal.")
+
+            requests = []
+            requests.append(
+                ("The capital of France is", SamplingParams(max_tokens=1),
+                 DisaggregatedParams(request_type="context_only")))
+
+            responses = send_requests_to_worker(requests, 0, intercomm)
+            output = responses[0]
+            print(f"Output: {output}")
+            assert output[0].text == "Paris" or output[0].text == " Paris"
+            print(f"Output: {output[0].disaggregated_params}")
+            assert output[0].disaggregated_params is not None
+            print(f"Output: {output[0].disaggregated_params.request_type}")
+            assert output[0].disaggregated_params.request_type == "context_only"
+
+            generation_request_disagg_params = output[0].disaggregated_params
+            generation_request_disagg_params.request_type = "generation_only"
+            requests = []
+            requests.append(
+                ("The capital of France is", SamplingParams(max_tokens=10),
+                 generation_request_disagg_params))
+
+            responses = send_requests_to_worker(requests, 1, intercomm)
+            output = responses[0]
+            assert output[0].text.startswith(
+                "Paris") or output[0].text.startswith(" Paris")
+
+            # Send a non-disaggregated request for output verification
+            requests = []
+            requests.append(("The capital of France is",
```

Comment:
Can we match the prompt used in multi-GPU tests: 

```
What is the capital of Germany?
```
And expected string:
```
The capital of Germany is Berlin
```

Reply:
Updated.